gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/FakeStorageRpc.java

```diff
@@ -117,18 +121,20 @@ public StorageObject create(StorageObject object, InputStream content, Map<Optio
       if (!so.getName().startsWith(prefix)) {
         continue;
       }
-      int nextSlash = so.getName().indexOf("/", prefix.length());
-      if (nextSlash >= 0) {
-        String folderName = so.getName().substring(0, nextSlash + 1);
-        if (folders.containsKey(folderName)) {
+      if (delimiter != null) {
+        int nextSlash = so.getName().indexOf(delimiter, prefix.length());
+        if (nextSlash >= 0) {
+          String folderName = so.getName().substring(0, nextSlash + 1);
+          if (folders.containsKey(folderName)) {
```

Comment:
Factored out into a helper function so we can use the early-exit pattern to reduce nesting. This makes the code arguably more readable (let me know if you don't like it since it's really a matter of taste at this point).